python/local_path_visualizer.py

```diff
@@ -221,63 +220,104 @@ def y(x):
+                                               round(state.speedKmph, LATLON_TEXT_DECIMAL_PLACES)), ha='center')
 
     while not rospy.is_shutdown():
         state = sailbot.getCurrentState()
-        referenceLatlon = nextGlobalWaypoint  # Ensure this matches createLocalPathSS referenceLatlon for best results
 
         # Convert values from latlon to XY, relative to the referenceLatlon
-        positionXY = utils.latlonToXY(state.position, referenceLatlon)
-        nextGlobalWaypointXY = utils.latlonToXY(nextGlobalWaypoint, referenceLatlon)
-        nextLocalWaypointXY = utils.latlonToXY(nextLocalWaypoint, referenceLatlon)
-        previousLocalWaypointXY = utils.latlonToXY(previousLocalWaypoint, referenceLatlon)
-        localPathXY = [utils.latlonToXY(localWaypoint, referenceLatlon) for localWaypoint in localPath]
-        localPathX = [xy[0] for xy in localPathXY]
-        localPathY = [xy[1] for xy in localPathXY]
-        shipsXY = obs.getObstacles(state, referenceLatlon)
-
-        _, isStartEast, slope, y_intercept = getPerpLine(previousLocalWaypointXY, nextLocalWaypointXY)
-        _, glob_isStartEast, glob_slope, glob_y = getPerpLine(previousGlobalWaypointXY, nextGlobalWaypointXY, True)
-
-        # Update plots
-        localPathPlot.set_xdata(localPathX)
-        localPathPlot.set_ydata(localPathY)
-        nextGlobalWaypointPlot.set_xdata(nextGlobalWaypointXY[0])
-        nextGlobalWaypointPlot.set_ydata(nextGlobalWaypointXY[1])
-        nextLocalWaypointPlot.set_xdata(nextLocalWaypointXY[0])
-        nextLocalWaypointPlot.set_ydata(nextLocalWaypointXY[1])
-        positionPlot.set_xdata(positionXY[0])
-        positionPlot.set_ydata(positionXY[1])
-        positionPlot.set_marker((3, 0, state.headingDegrees - 90))  # Creates a triangle with correct 'heading'
-        positionPlotTail.set_xdata(positionXY[0] - 1 * math.cos(math.radians(state.headingDegrees)))
-        positionPlotTail.set_ydata(positionXY[1] - 1 * math.sin(math.radians(state.headingDegrees)))
-        positionPlotTail.set_marker((5, 0, state.headingDegrees - 90))  # Creates a tail for the visualizer
-
-        lineX, lineY = getPerpPlot(isStartEast, slope, y_intercept, nextLocalWaypointXY)
-        waypointReachedPlot.set_xdata(lineX)
-        waypointReachedPlot.set_ydata(lineY)
-
-        glob_X, glob_Y = getPerpPlot(glob_isStartEast, glob_slope, glob_y, nextGlobalWaypointXY)
-        globalWaypointReachedPlot.set_xdata(glob_X)
-        globalWaypointReachedPlot.set_ydata(glob_Y)
-
-        # Resize axes if needed
-        if needAxesResized(positionXY, nextGlobalWaypointXY, xPLim, xNLim, yPLim, yNLim):
-            xPLim, xNLim, yPLim, yNLim = getXYLimits(positionXY, nextGlobalWaypointXY)
-            axes.set_xlim(xNLim, xPLim)
-            axes.set_ylim(yNLim, yPLim)
+        if nextGlobalWaypoint:
+            referenceLatlon = nextGlobalWaypoint  # Matches createLocalPathSS referenceLatlon for best results
```

Comment:
Currently, the plot the appears right away looks like this
![image](https://user-images.githubusercontent.com/65756895/127717831-a86f1524-e719-4462-9521-0ec9be4431b9.png)
Since `shipsXY` uses `referenceLatlon`, which is equal  to `nextGlobalWaypoint`, the ships do not appear in the initial plot. Is this the desired behavior?